src/utils/provider.ts

```diff
@@ -14,9 +14,11 @@ export async function getProviderRegistry() {
 
   return createProviderRegistry({
     openai: createOpenAI({
+      baseURL: config?.providersConfig?.openai.baseURL ?? 'https://api.openai.com/v1',
       apiKey: config?.providersConfig?.openai.apiKey,
     }),
     deepseek: createDeepSeek({
+      baseURL: config?.providersConfig?.deepseek.baseURL ?? 'https://api.deepseek.com/v1',
```

Comment:
seems duplicated code as the switch logic in the component file. need to change the code both side when change something in the future.